kedro/io/kedro_data_catalog.py

```diff
@@ -264,6 +307,11 @@ def list(self, regex_search: str | None = None) -> list[str]:
         return [ds_name for ds_name in self._datasets if pattern.search(ds_name)]
 
     def save(self, name: str, data: Any) -> None:
+        # TODO: remove when removing old catalog
+        """Save data to a registered dataset."""
+        self.save_data(name, data)
+
+    def save_data(self, name: str, data: Any) -> None:
```

Comment:
I am not sure of what is the rationale of keeping ``load_data`` and ``save_data``, but I see this was decided in a TD. Could ou juste reexplain the expected benefit? (I suspect that we keep ``load_data`` to deal with version ; and we keep ``save_data`` for "symmetry")

Reply:
You're right about `load_data,` and we are keeping both for now to ensure compatibility with the old catalog. Version will be probably moved from the catalog methods interface soon and technically we won't need `load_data` and `save_data` as we can replace them with `catalog["dataset_name"].load()` and `catalog["dataset_name"].save()`.

Reply:
We had one more discussion on this and decided that we will probably keep `load` and `save` at the catalog level, so framework components interact with the datasets via catalog abstraction.

Reply:
Yes, I was confused by the renaming but it makes more sense like this 👍